File: iOS-LazyTableView/iOS-LazyTableView/L3SDK/Utility/L3SDKUtilityLazyLoad.m
//
//  L3SDKUtilityLazyLoad.m
//  iOS-LazyTableView
//
//  Created by Domenico Vacchiano on 15/07/15.
//  Copyright (c) 2015 DomenicoVacchiano. All rights reserved.
//

#import "L3SDKUtilityLazyLoad.h"


@implementation L3SDKUtilityLazyLoad
@synthesize currentCursor;
@synthesize isEnabled;
@synthesize pageSize;

-(id)initWithPageSize:(int)pSize andIsEnabled:(BOOL)enabled{
    
    self=[super init];
    if (self) {
        self.pageSize=pSize;
        self.isEnabled=enabled;
    }
    return self;
    
}
-(int)nextCursor{
    
    if (!self.isEnabled) {
        return -1;
    }
    
    if (self.currentCursor==-1) {
        currentCursor= 0;
    }
    else if (self.currentCursor==0) {
        currentCursor=self.pageSize;
    }else{
        currentCursor+=self.pageSize;
    }
    return self.currentCursor;

}

-(void)resetCursor{
    currentCursor=-1;
}

@end

File: iOS-LazyTableView-Swift/iOS-LazyTableView-Swift/L3SDKUtilityLazyLoad.swift
//
//  L3SDKUtilityLazyLoad.swift
//  iOS-LazyTableView
//
//  Created by Jared on 5/27/16.
//  Copyright © 2016 DomenicoVacchiano. All rights reserved.
//

import Foundation

class L3SDKUtilityLazyLoad
{
    var currentCursor: Int = -1
    var enabled: Bool = true
    var pageSize: Int = -1
    
    init()
    {}
    
    init(withPageSize pSize: Int, andEnabled isEnabled: Bool)
    {
        self.pageSize = pSize
        self.enabled = isEnabled
    }
    
    func nextCursor() -> Int
    {
        if !self.enabled { return -1 }
        
        switch currentCursor
        {
            case -1:
                currentCursor = 0
                break
            case 0:
                currentCursor = self.pageSize
                break
            default:
                currentCursor += pageSize
        }
        return currentCursor
    }
    
    func resetCursor()
    {
        currentCursor = -1
    }
}